src/Featurizers/ShortGrainDropperFeaturizer.h

```diff
@@ -0,0 +1,255 @@
+// ----------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License
+// ----------------------------------------------------------------------
+#pragma once
+
+#include "../Featurizer.h"
+#include "../Archive.h"
+
+namespace Microsoft {
+namespace Featurizer {
+namespace Featurizers {
+
+
+/////////////////////////////////////////////////////////////////////////
+///  \class         ShortGrainDropperTransformer
+///  \brief         Return true if a grain needs dropping
+///
+class ShortGrainDropperTransformer : public StandardTransformer<std::vector<std::string>, bool> {
+public:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Types
+    // |
+    // ----------------------------------------------------------------------
+    using BaseType                          = StandardTransformer<std::vector<std::string>, bool>;
+    using GrainsSet                         = std::unordered_set<std::string>;
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    explicit ShortGrainDropperTransformer(GrainsSet grainsToDrop);
+    explicit ShortGrainDropperTransformer(Archive &ar);
+
+    ~ShortGrainDropperTransformer(void) override = default;
+
+    FEATURIZER_MOVE_CONSTRUCTOR_ONLY(ShortGrainDropperTransformer);
+
+    bool operator==(ShortGrainDropperTransformer const &other) const;
+
+    void save(Archive &ar) const override;
+
+private:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Data
+    // |
+    // ----------------------------------------------------------------------
+    GrainsSet const                         _grainsToDrop;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Methods
+    // |
+    // ----------------------------------------------------------------------
+     // MSVC has problems when the declaration and definition are separated
+    void execute_impl(typename BaseType::InputType const &input, typename BaseType::CallbackFunction const &callback) override {
+        callback(_grainsToDrop.find(Traits<decltype(input)>::ToString(input)) != _grainsToDrop.end());
+    }
+};
+
+/////////////////////////////////////////////////////////////////////////
+///  \class         ShortGrainDropperEstimator
+///  \brief         Estimator to determine which grain to drop
```

Comment:
>Estimator to determine which grain to drop [](start = 20, length = 42)

This comment can be improved. Based on the ctor parameters, I gather that the logic used to determine if a grain should be dropped is based on time. However, I don't see anything in the input related to time.

Reply:
it looks like the way to determine which grain to drop is not directly based on time